src/elements/element.line.js

```diff
@@ -135,8 +151,173 @@ function fastPath(ctx, points, spanGaps) {
 	}
 }
 
-function useFastPath(vm) {
-	return vm.tension === 0 && !vm.steppedLine && !vm.fill && !vm.borderDash.length;
+function useFastPath(vm, options) {
+	return !vm.tension && !vm.steppedLine && !vm.fill && (!vm.borderDash || !vm.borderDash.length) && !options.bounds;
+}
+
+function angleBetween(a, s, e) {
+	return a === s || a === e ||
+		(angleDiffCW(a, s) > angleDiffCW(a, e) && angleDiffCW(s, a) < angleDiffCW(e, a));
+}
+
+function axisFn(axis) {
+	if (axis === 'r') {
+		return {
+			between: angleBetween,
+			compare: angleDiff,
+			normalize: normalizeAngle,
+		};
+	}
+	return {
+		between: (n, s, e) => n >= s && n <= e,
+		compare: (a, b) => a - b,
+		normalize: x => x
+	};
+}
+
+function makeSubSegment(start, end, loop, count) {
+	return {
+		start: start % count,
+		end: end % count,
+		loop: loop && (end - start + 1) % count === 0
+	};
+}
+
+function getSegment(segment, points, bounds) {
+	const {axis, start: startBound, end: endBound} = bounds;
+	const {between, normalize} = axisFn(axis);
+	const count = points.length;
+	let {start, end, loop} = segment;
+	let i, ilen;
+
+	if (loop) {
+		start += count;
+		end += count;
+		for (i = 0, ilen = count; i < ilen; ++i) {
+			if (!between(normalize(points[start % count]._view[axis]), startBound, endBound)) {
+				break;
+			}
+			start--;
+			end--;
+		}
+		start %= count;
+		end %= count;
+	}
+
+	if (end < start) {
+		end += count;
+	}
+	return {start, end, loop};
+}
+
+export function boundSegment(segment, points, bounds) {
```

Comment:
Can you add a doc to this method explaining its purpose and maybe to `boundSegments` below as well?

Reply:
done